homeassistant/components/synology_dsm/config_flow.py

```diff
@@ -250,7 +252,13 @@ async def async_step_init(self, user_input=None):
                     default=self.config_entry.options.get(
                         CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL
                     ),
-                ): cv.positive_int
+                ): cv.positive_int,
+                vol.Optional(
+                    CONF_SECURITY,
+                    default=self.config_entry.options.get(
+                        CONF_SECURITY, DEFAULT_SECURITY
+                    ),
+                ): bool,
```

Comment:
Agree with Nick. This should not be an option but just disabled by default.

Reply:
So you are ok to make a request to fetch security data even if we don't use it ?

Reply:
Searching to not make that with `entity_registry_enabled_default` ...

Reply:
Ideally the update function is smart enough to know that if it has no subscribers listening for the data that is should skip requesting those data sets.

Reply:
Please see

https://github.com/ProtoThis/python-synology/blob/674dfb3c943e464680c33c8bc4975139d8c72f5c/synology_dsm/synology_dsm.py#L258-L314

To think what I can do

Reply:
Looks like that will do the trick.  Ideally properties never do I/O, but that is an existing issue.

It also means you can use `DataUpdateCoordinator` in a future PR since the underlying package now has a single entry point to trigger and update.
https://developers.home-assistant.io/docs/integration_fetching_data/#coordinated-single-api-poll-for-data-for-all-entities

Reply:
What will do the trick ?
The UI option ?
The coordinator ?

Reply:
I may found something but cannot be used for the `storage` API, will show you just after some tests

Reply:
> What will do the trick ?

The upstream package change to `python-synology`

Reply:
This integration uses `python-synology` 
https://github.com/home-assistant/core/blob/70b29dc823dd8929634c51dd4af16f643632a6cc/homeassistant/components/synology_dsm/manifest.json#L5

Reply:
https://github.com/ProtoThis/python-synology/blob/674dfb3c943e464680c33c8bc4975139d8c72f5c/synology_dsm/synology_dsm.py#L258-L314

I wasn't clear that this is the existing code and not proposed new code.

Looking at it, it seems that if you never call `def security` and call `def update` it will not poll the security data because `self._security` is `None`.  

If you had a `DataUpdateCoordinator` instance for each `utilisation`, `storage`, and `security` it would take care of requesting only if the entities are actually added since it does not poll if there are no entities listening for updates.

Reply:
Look at last commit of https://github.com/Quentame/home-assistant/commits/synology_dsm/disable-entries-and-not-fetch 

I loose all `utilisation` and `security` sensors

Reply:
> If you had a `DataUpdateCoordinator` instance for each `utilisation`, `storage`, and `security` it would take care of requesting only if the entities are actually added since it does not poll if there are no entities listening for updates.

Seeing your message just now

Reply:
Yes it's now clear

Reply:
If I turn off the security sensor `self._security` will have already have been set here https://github.com/ProtoThis/python-synology/blob/674dfb3c943e464680c33c8bc4975139d8c72f5c/synology_dsm/synology_dsm.py#L297 so it going to continue to poll here https://github.com/ProtoThis/python-synology/blob/674dfb3c943e464680c33c8bc4975139d8c72f5c/synology_dsm/synology_dsm.py#L264

Your original approach of the option to turn on/off the sensor does eliminate this issue.  The downside is that every new sensor type would have to be added to the options flow which will make it a lot more difficult to refactor later to tie specific sensors to specific api calls to avoid the ones that are disabled.

Reply:
@bdraco Yes, but in the case we have one `DataUpdateCoordinator` per API:

When requesting a `storage` fetch (and with a `security` coordinator somewhere) the lib `update` will fetch **storage** + **security**

After, the security `DataUpdateCoordinator` will also update with fetch **storage** + **security**.

So it doubles requests !

OR,

Every `DataUpdateCoordinator` should have its own instance of `SynoApi` or `SynologyDSM`, it will do even more requests and will login to the NAS for every `DataUpdateCoordinator`, I beat it will have an error at login (logged multiple times)

Reply:
You would need to some refactoring remove the the `update` and only have the coordinator instances for each endpoint share the single `SynologyDSM` so you aren't logging in multiple times.
There is a good example on how to accomplish this in `nws/__init__.py:async_setup_entry`

Reply:
> If I turn off the security sensor `self._security` will have already have been set here https://github.com/ProtoThis/python-synology/blob/674dfb3c943e464680c33c8bc4975139d8c72f5c/synology_dsm/synology_dsm.py#L297 so it going to continue to poll here https://github.com/ProtoThis/python-synology/blob/674dfb3c943e464680c33c8bc4975139d8c72f5c/synology_dsm/synology_dsm.py#L264

Yes totally, setting the sensor to off will make the API continue to fetch on it.

> Your original approach of the option to turn on/off the sensor does eliminate this issue. The downside is that every new sensor type would have to be added to the options flow which will make it a lot more difficult to refactor later to tie specific sensors to specific api calls to avoid the ones that are disabled.

Not totally, one option per API, not per sensor.


Maybe, at `SynoApi.async_update` I can check all entities of this config, and if all entities of a specific API are disabled: something like this `self.dsm._security = None`.

@bdraco Do you know how to get all entities of one config ?

Reply:
Like this:

```python
registry = await entity_registry.async_get_registry(hass)
    conditions: List[Dict[str, Any]] = []

    # Get all the integrations entities for this device
    for entry in entity_registry.async_entries_for_device(registry, device_id):
```

I suppose

Reply:
Generally in this case we would want each entity to subscribe and then we can look at the list of subscribers and determine which endpoints to poll.  `august/subscriber.py` has an example